### DISCUSSION TITLE
Allow blocking PR merge while a review is in progress

### DISCUSSION THREAD

**Discussion by hakamadare** [community/community#34607] [Pull Requests]
Here's the situation I routinely encounter:
* I start reviewing a colleague's PR
* While I am drafting my review, another colleague completes a review and approves the PR
* By the time I have submitted my review, the PR has already been merged

Yes, there are a couple of ways I could work around this, such as:
* put a label of "I AM REVIEWING DO NOT MERGE" on the PR before starting my review
* post a very quick review that requests changes and says "wait for me to finish the review"

Both of those are clunky IMO, and depend on me remembering to do a thing in a particular order.  Here's the behavior I'd like to see instead:
* As soon as I click "Start a review", set a status on the PR of "Review in progress"
* Allow me to configure a branch protection rule that says that PRs cannot be merged while they have reviews in progress
* Allow the submitter to dismiss/override my review in progress, using the existing mechanism for dismissing a review (prompt for an explanation etc.)

**Comment by ScottLangham** [comment]
This!! With one additional thought. When doing a review and I create a comment, make that comment IMMEDIATELY visible to the author.

**Comment by SharakPL** [comment]
Making labels block merging is what I'm looking for https://github.com/community/community/discussions/40927
It's certainly more useful than blocking every time someone starts a review.

**Comment by GabyCT** [comment]
For branch protection rules you can see https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule and https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request

**Reply by SharakPL** [reply]
Yeah, but there's not enough of these rules. Is there a way to block merging based on specific label set on the PR?

**Comment by lrworth** [comment]
Even something passive would be useful, like an indicator next to people in the "Reviewers" list who have interacted with the review tab (e.g. make a comment or clicked "Viewed").

**Comment by brandonthomas** [comment]
I was just about to submit this same issue when I found this. I find this issue incredibly frustrating and disheartening when I am putting together thoughtful feedback and frankly it makes me hate working in GitHub. Here's my $0.02 on this issue:

>As a reviewer, I am often frustrated that I sometimes work more slowly than others when looking at changes. Maybe I'm more thorough or maybe I'm just slow. Either way, due to this I often hit sequencing issues. I might have a review in progress and the author gets sign off from someone else and merges. Basically, my entire PR is now worthless as the likelihood of a person starting a new PR and incorporating that feedback is slim. What I'm asking for is some sort of way for authors to know that they have people still looking at their submission. This way, everyone, regardless as to speed, can be heard. Imagine the PR reviewer is using a screen reader or voice to text. Does their feedback matter less because they might not be as fast as others? Ultimately this should increase code quality too. Maybe this is a bigger issue on larger code bases but I've hit this issue twice just this week and it's not the first time. Maybe a dialog that says "Are you sure you want to merge? 3 people have in progress reviews [some list of their names so you can follow up]".

**Comment by ddh** [comment]
While a review is in progress, as the PR owner, it would be nice to see where the reviewer is at in their progress, in terms of number of files viewed. This could be a useful metric to see and a way to view a reviewer's progress without having to bug them about it. 

![image](https://user-images.githubusercontent.com/9056483/225378221-916c3bae-3911-4639-8c27-5adeb978bb52.png)

**Comment by digitalpacman** [comment]
Recently I found a bug and started a review.  I became distracted and forgot to submit my review.  In my head thought I had.  The code was later merged, and, caused the company to lose money with a bug present that I had commented on.  This 100% needs to be a feature.
